### PR TITLE
#26 view connection requests

### PR DESCRIPTION
**Pull request by Ghaith-Chrit**
Added functionality to view/accept/reject connection requests

Closes: #26

### CODE REVIEW

**Review by JamesPartsafas — APPROVED** [review]
Backend looks good

File: synapsis/src/main/resources/templates/pages/network.html [hunk 1]
```diff
@@ -40,51 +40,23 @@
                 </div>
                 <div class="overflow-x-auto whitespace-nowrap h-80">
 
-                    <div class="inline-block rounded-2xl shadow w-56 h-72 bg-white mt-4 mr-4">
+                    <p th:text="${error}"></p>
+                    <div th:each="pendingConnectionRequest: ${pendingConnectionRequests}" class="inline-block rounded-2xl shadow w-56 h-72 bg-white mt-4 mr-4">
                         <img class="w-56 h-40 rounded-t-2xl">
-                        <div class="font-sans text-base font-light mt-2 ml-4">Tanya Leduc</div>
+                        <div class="font-sans text-base font-light mt-2 ml-4" th:text="${pendingConnectionRequest.name}">Tanya Leduc</div>
```
Comment (CHADBUCK13) [inline synapsis/src/main/resources/templates/pages/network.html:46]: We should remove the innerHTML here 'Tanya Leduc"

File: synapsis/src/test/java/com/soen/synapsis/integration/ConnectionRepositoryTest.java [hunk 2]
```diff
@@ -73,4 +74,20 @@ void itShouldNotFindAcceptedConnectionsByReceiverIDIfUserDoesNotExist() {
         assertTrue(result.isEmpty());
     }
 
+    @Test
+    void itShouldFindPendingConnectionsByReceiverID() {
```
Comment (CHADBUCK13) [inline synapsis/src/test/java/com/soen/synapsis/integration/ConnectionRepositoryTest.java:78]: should simplify to PendingConnectionsByReceiverID_Success or something simple like this

File: synapsis/src/test/java/com/soen/synapsis/integration/ConnectionRepositoryTest.java [hunk 3]
```diff
@@ -73,4 +74,20 @@ void itShouldNotFindAcceptedConnectionsByReceiverIDIfUserDoesNotExist() {
...
+        ConnectionKey connectionKey = new ConnectionKey(appUser1.getId(), appUser2.getId());
+        Connection connection = new Connection(connectionKey, appUser1, appUser2, true);
+        underTest.save(connection);
+        assertTrue(underTest.findPendingConnectionsByReceiverID(appUser2.getId()).size() == 1);
+    }
+
+
+    @Test
+    void itShouldNotFindPendingConnectionsByReceiverIDIfConnectionIsNotPending() {
```
Comment (CHADBUCK13) [inline synapsis/src/test/java/com/soen/synapsis/integration/ConnectionRepositoryTest.java:87]: same as above

File: synapsis/src/test/java/com/soen/synapsis/unit/appuser/connection/ConnectionServiceTest.java [hunk 4]
```diff
@@ -139,4 +140,74 @@ void getConnections() {
         assertEquals(allConnections, returnValue);
     }
 
+    @Test
+    void rejectConnectionWithNonexistentConnection() {
```
Comment (CHADBUCK13) [inline synapsis/src/test/java/com/soen/synapsis/unit/appuser/connection/ConnectionServiceTest.java:144]: typo, should be: NonExistent

File: synapsis/src/test/java/com/soen/synapsis/unit/appuser/connection/ConnectionServiceTest.java [hunk 5]
```diff
@@ -139,4 +140,74 @@ void getConnections() {
...
+        when(connectionRepository.findById(any(ConnectionKey.class))).thenReturn(Optional.of(connection));
+        ArgumentCaptor<ConnectionKey> connectionArgumentCaptor = ArgumentCaptor.forClass(ConnectionKey.class);
+        verify(connectionRepository, atMostOnce()).deleteById(connectionArgumentCaptor.capture());
+
+        AppUserDetails appUserDetails = new AppUserDetails(candidateUser1);
+        assertEquals(underTest.rejectConnection(appUserDetails, candidateUser2.getId()), "redirect:/network");
+    }
+
+    @Test
+    void acceptConnectionWithNonexistentConnection() {
```
Comment (CHADBUCK13) [inline synapsis/src/test/java/com/soen/synapsis/unit/appuser/connection/ConnectionServiceTest.java:179]: Same as above

**Review by CHADBUCK13 — CHANGES_REQUESTED** [review]
Looks good, just couple small changes

**Review by CHADBUCK13 — APPROVED** [review]

**Pull request merged**